### PR TITLE
Rkhan/tests and refactoring

### LINKED ISSUE #1: InputField Molecule has duplicate code
Hi Jad,

First of all, great repo man, I really learnt alot from it. I was getting into Typescipt in Vue and this repo hit the button.
 
About the above issue, don't you think using a `<slot>` would have been better, specifically for the InputField.molecule file?

I feel that you are duplicating the props (just to pass them again to the input component).

Again, thanks for making this public, its a great learning opportunity.

### LINKED ISSUE #4: Add tests for core components
These are the remaining core components left to test:

1. BaseButton
2. BaseIcon
3. BaseText

### PR DESCRIPTION
**Pull request by RizaHKhan**
#4 
#1 

Sorry this is such a large PR :(. Kill two birds with one stone.

### CODE REVIEW

**Comment by JadRizk** [comment]
Hey man ! Sure don't worry about the PR size 😅

Okay so the idea of having a `baseInput` component is have a common input across the whole application(input field, checkbox, ...)

I would suggest keeping the BaseInput as it was - simple input (atom) - and build a new component called a `textField` component that could handle a the Input, Label and, error tag.

**Comment by RizaHKhan** [comment]
@JadRizk Agreed!

BaseInput reverted to original version and added TextField component (also an atom, though it has a little bit more functionality).

Also, you will notice I added a `StyleGuide` file where we can showcase all of our various components and how they look. I picked this up from my current place of work and I find it quite useful. I actually found an issue that I've added as an issue.

Also added more tests :)

Let me know if its all good, thanks!

File: tests/unit/TextField.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { shallowMount } from '@vue/test-utils'
+import TextField from '@/components/atoms/TextField.atom.vue'
+
+describe('TextField', () => {
+  const id = 'greatestId'
+  const name = 'GhenghisKhan'
```
Comment (JadRizk) [inline tests/unit/TextField.spec.ts:6]: Consider adding the following fields to remove existing warnings

```suggestion
  const id = 'greatestId'
  const name = 'GhenghisKhan'
  const label = 'TextField-Label'
  const iconName = 'IconName'
  const modelValue = 'modelValue'
```

File: tests/unit/TextField.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  it('should emit a value', async () => {
+    const wrapper = shallowMount(TextField, {
+      props: { id, name },
+    })
```
Comment (JadRizk) [inline tests/unit/TextField.spec.ts:11]:
```suggestion
   const wrapper = shallowMount(TextField, {
      props: { id, name, label, iconName, modelValue },
    })
```

File: tests/unit/TextField.spec.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    wrapper.vm.$emit('update:modelValue', '123')
+    await wrapper.vm.$nextTick()
+
+    expect(wrapper.emitted('update:modelValue')).toBeTruthy()
+    expect(wrapper.emitted('update:modelValue')[0]).toEqual(['123'])
+  })
+
+  it('emits on-blur when on blur', async () => {
+    const wrapper = shallowMount(TextField, {
+      props: { id, name },
```
Comment (JadRizk) [inline tests/unit/TextField.spec.ts:21]:
```suggestion
      props: { id, name, label, iconName, modelValue },
```

**Review by JadRizk — CHANGES_REQUESTED** [review]

File: src/components/atoms/TextField.atom.vue [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
-      :type="type"
       :placeholder="placeholder"
+      class="default-input"
+      :class="inputWrapperClass"
+      @blur="handleOnBlur"
       :value="modelValue"
-      :errorMode="validationStatus.$error"
-      @on-blur="handleOnBlur"
-      @update:modelValue="emit('update:modelValue', $event)"
+      @input="emit('update:modelValue', $event.target.value)"
```
[the remark this answers is not in the document]
Comment (RizaHKhan) [reply src/components/atoms/TextField.atom.vue]: reverted to original version. When I was creating it, I felt that the original version seemed better anyways.

File: tests/unit/TextField.spec.ts (hunk 1 above)
Comment (RizaHKhan) [reply tests/unit/TextField.spec.ts]: props added

**Comment by RizaHKhan** [comment]
There is an issue in th test:

image
Since we are using aliases for our imports, its throwing the jest tests off. I've searched high and low for a resolution but nothing. Perhaps you would know?

**Review by JadRizk — APPROVED** [review]

**Pull request merged**